spec/commands/newgem_spec.rb

```diff
@@ -158,22 +158,32 @@ def gem_skeleton_assertions(gem_name)
         expect(bundled_app("test_gem/README.md").read).not_to include("github.com/bundleuser")
       end
     end
+  end
 
-    context "git is not installed in the system" do
-      before do
-        FileUtils.rm ENV["GIT_CONFIG"].to_s if File.exist?(ENV["GIT_CONFIG"])
-        reset!
-        in_app_root
-        bundle "gem #{gem_name}"
-        remove_push_guard(gem_name)
-      end
+  it "creates a new git repository" do
+    in_app_root
+    bundle "gem test_gem"
+    expect(bundled_app("test_gem/.git")).to exist
+  end
 
-      it "contribute URL set to [USERNAME]" do
-        expect(bundled_app("test_gem/README.md").read).to include("[USERNAME]")
-        expect(bundled_app("test_gem/README.md").read).not_to include("github.com/bundleuser")
-      end
+  context "when git is not avaiable" do
+    let(:gem_name) { "test_gem" }
 
-      it_should_behave_like "git config is absent"
+    before do
+      allow(Bundler).to receive(:git_present?).and_return(false)
```

Comment:
I tried looking into this quickly, but the biggest problem was that the `ruby` binary was in the same part of `$PATH` along with `git`.